### PR TITLE
[NPU] add _npu_identity op and api, test=develop

### PR DESCRIPTION
**Pull request by qili93**
### PR types

Others

### PR changes

OPs

### Describe

Add NPU identity op for Ascend CANN OP Identity.

This op and api will be used to transform parameters of BatchNorm to improve performance on Ascend NPU, details in https://github.com/PaddlePaddle/Paddle/pull/48039 which is still working in progress (Draft status now).

Also previous PR closed and npu_identity infer meta changed: https://github.com/PaddlePaddle/Paddle/pull/47688

### CODE REVIEW

File: python/paddle/tensor/manipulation.py [hunk 1]
```diff
@@ -4853,3 +4853,66 @@ def index_add_(x, index, axis, value, name=None):
...
     setattr(core.eager.Tensor, name, func)
+
+
+# TODO(qili93): remove this op after custom op and custom device
+# integrated and then move this op along with its code to plugin.
+def _npu_identity(x, format=-1):
+    """
+
+    This OP takes in the Tensor :attr:`x` and change it to ouptut with
+    aclFormat with int value. Thsi API is only used for Ascend NPU.
```
Comment (zyfncg) [inline python/paddle/tensor/manipulation.py]: Thsi -> This

File: python/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -4853,3 +4853,66 @@ def index_add_(x, index, axis, value, name=None):
 for name, func in __METHODS.items():
     setattr(core.VarBase, name, func)
     setattr(core.eager.Tensor, name, func)
+
+
+# TODO(qili93): remove this op after custom op and custom device
+# integrated and then move this op along with its code to plugin.
+def _npu_identity(x, format=-1):
```
Comment (zyfncg) [inline python/paddle/tensor/manipulation.py]: 这个 api 后面要去掉的话可能放在 `python/paddle/incubate` 下面更合适一些

File: python/paddle/tensor/manipulation.py (hunk 1 above)
Comment (qili93) [reply python/paddle/tensor/manipulation.py]: fixed

File: python/paddle/tensor/manipulation.py (hunk 2 above)
Comment (qili93) [reply python/paddle/tensor/manipulation.py]: 已修改，API 迁移到 incubate

File: python/paddle/incubate/tensor/manipulation.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zyfncg) [inline python/paddle/incubate/tensor/manipulation.py]: 2021->2022

**Review by zyfncg — DISMISSED** [review]
LGTM

File: python/paddle/incubate/__init__.py [hunk 4]
```diff
@@ -53,4 +54,5 @@
     'segment_max',
     'segment_min',
     'identity_loss',
+    '_npu_identity',
```
Comment (jeff41404) [inline python/paddle/incubate/__init__.py]: if `_npu_identity`  is used temporarily and will be deleted later, so not need to add in \_\_all\_\_

File: python/paddle/incubate/tensor/manipulation.py (hunk 3 above)
Comment (qili93) [reply python/paddle/incubate/tensor/manipulation.py]: fixed

File: python/paddle/incubate/__init__.py (hunk 4 above)
Comment (qili93) [reply python/paddle/incubate/__init__.py]: fixed

**Review by jeff41404 — APPROVED** [review]

**Pull request merged**